### PR TITLE
Handle land use change uptake fluxes correctly

### LINKED ISSUE #523: Negative land use change emissions in Hector v3
So some of the ssps have negative land-use change emissions in Hector v3 this should throw an error message and the negative emissions need to be incorporated into the carbon cycle via land use uptake. 

image

```{r}
remotes::install_github("jgcri/hectordata@master")

devtools::load_all()
library(magrittr)
library(ggplot2)

ini <- system.file("input/hector_ssp119.ini", package = "hectordata") 
core <- newcore(ini)
run(core)
out1 <- fetchvars(core, 1900:2100, vars = c(GLOBAL_TEMP(), LUC_EMISSIONS()))
ggplot(data = out1) +
    geom_line(aes(year, value)) +
    geom_hline(yintercept = 0) +
    facet_wrap("variable", scales = "free") + 
    theme_bw() + 
    labs(title = "SSP119 Hector Results")
  
```

```
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.5  hector_3.0.0   magrittr_2.0.1 testthat_3.0.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7        compiler_3.6.3    pillar_1.6.3      prettyunits_1.1.1 remotes_2.4.0    
 [6] tools_3.6.3       digest_0.6.28     pkgbuild_1.1.0    pkgload_1.1.0     memoise_1.1.0    
[11] lifecycle_1.0.1   tibble_3.1.5      gtable_0.3.0      pkgconfig_2.0.3   rlang_0.4.11     
[16] DBI_1.1.0         cli_3.0.1         rstudioapi_0.13   yaml_2.2.1        dplyr_1.0.7      
[21] withr_2.4.2       generics_0.1.0    vctrs_0.3.8       desc_1.2.0        fs_1.5.0         
[26] devtools_2.3.2    tidyselect_1.1.1  rprojroot_2.0.2   grid_3.6.3        glue_1.4.2       
[31] R6_2.5.1          processx_3.4.5    BH_1.75.0-0       fansi_0.5.0       sessioninfo_1.1.1
[36] farver_2.1.0      purrr_0.3.4       callr_3.5.1       scales_1.1.1      ps_1.5.0         
[41] ellipsis_0.3.2    usethis_1.6.3     assertthat_0.2.1  colorspace_2.0-2  labeling_0.4.2   
[46] utf8_1.2.2        munsell_0.5.0     crayon_1.4.1     
```

### PR DESCRIPTION
**Pull request by bpbond**
This PR
* Adds internal data structures and data-reading code for LUC_UPTAKE, paralleling LUC_EMISSIONS; add references to this in the INI files
* Change the LUC variables to fluxpools; the model will now stop with an error if negative values are given to either LUC emissions or uptake
* Use this new data time series in Hector computations
* Allow access to LUC_UPTAKE via the R package 
* Update comparison data. Because we were previously ignoring the LUC_UPTAKE columns in the emiss-constraints files, scenarios with 21st/22nd century land use change uptake fluxes diverge from previous outputs (see below)

ini

Closes #523

### CODE REVIEW

File: misc/main-api.cpp [hunk 1]
```diff
@@ -191,6 +191,8 @@ void read_and_set_co2(double tstrt, double tend, Core &core, istream &sim_gcam_e
                              message_data(t, unitval(daccs, U_PGC_YR)));
             core.sendMessage(M_SETDATA, D_LUC_EMISSIONS,
                              message_data(t, unitval(luc, U_PGC_YR)));
+            core.sendMessage(M_SETDATA, D_LUC_UPTAKE,
+                             message_data(t, unitval(luc, U_PGC_YR)));
```
Comment (kdorheim) [inline misc/main-api.cpp:195]: Not related to this work but all of the inputs need to be added to the misc/main-api?

File: src/rcpp_constants.cpp [hunk 2]
```diff
@@ -1586,6 +1586,13 @@ String LUC_EMISSIONS() {
 return D_LUC_EMISSIONS;
 }
 
+//' @rdname carboncycle
+//' @export
```
Comment (kdorheim) [inline src/rcpp_constants.cpp:1590]: Add @describeIn section?

**Review by kdorheim — APPROVED** [review]
One question and one request to add a comment otherwise looks great! 👍

File: misc/main-api.cpp (hunk 1 above)
Comment (bpbond) [reply misc/main-api.cpp:195]: Yes, they will. I added this mostly as a "don't forget" line (I've never built the GCAM API target myself).

File: src/rcpp_constants.cpp (hunk 2 above)
Comment (bpbond) [reply src/rcpp_constants.cpp:1590]: I'm not an expert but the [documentation](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html) says "use **either** @rdname or @describeIn to control where method documentation goes". And everything here as `@rdname` tags so I think we're good?

**Comment by kdorheim** [comment]
sounds good!

**Comment by bpbond** [comment]
Not sure what's going on with the lint failure, but I don't think it's me. OK to merge even if it's erroring @kdorheim ?

**Comment by kdorheim** [comment]
yes sounds good to me especially since this is not directly being merged into main

**Pull request merged**